Include/internal/pycore_immutability.h

```diff
@@ -0,0 +1,22 @@
+#ifndef Py_FREEZE_H
```

Comment:
Should this header also be renamed to `pycore_freeze.h?`

Reply:
As discussed I think we'll keep it as is for now to avoid any confusion with the deep freeze functionality, but wait and see what feedback we receive.